behave_django/management/commands/behave.py

```diff
@@ -1,56 +1,81 @@
+if LooseVersion('2.0') > LooseVersion('.'.join((str(x) for x in DJANGO_VERSION[:3]))):  # noqa: E501
+    from argparse import ArgumentParser
+
+    class CommandParser(ArgumentParser):  # noqa: F811
+        """
+        Customized ArgumentParser class to improve some error messages and
+        prevent SystemExit in several occasions, as SystemExit is unacceptable
+        when a command is called programmatically.
+        """
+        def __init__(self, **kwargs):
+            self.missing_args_message = kwargs.pop('missing_args_message', None)  # noqa: E501
+            self.called_from_command_line = kwargs.pop('called_from_command_line', None)  # noqa: E501
+            kwargs.pop('allow_abbrev', False)
+            super(CommandParser, self).__init__(**kwargs)
+
+        def parse_args(self, args=None, namespace=None):
+            # Catch missing argument for a better error message
+            if (self.missing_args_message and
+                    not (args or any(not arg.startswith('-') for arg in args))):  # noqa: E501
+                self.error(self.missing_args_message)
+            return super().parse_args(args, namespace)
+
+
+def get_command_line_option(argv, *args, **kwargs):
     """
-    Additional command line arguments for the behave management command
+    Return the value of a command line option (which should include leading
+    dashes, e.g. '--testrunner') from an argument list. Return None if the
+    option wasn't passed or if the argument list couldn't be parsed.
     """
-    parser.add_argument(
-        '--noinput',
-        '--no-input',
-        action='store_const',
-        const=False,
-        dest='interactive',
-        help='Tells Django to NOT prompt the user for input of any kind.',
-    )
-    parser.add_argument(
-        '--failfast', action='store_const', const=True, dest='failfast',
-        help=('Tells Django to stop running the '
-              'test suite after first failed test.'),
-    )
-    parser.add_argument(
-        '-r', '--reverse', action='store_const', const=True, dest='reverse',
-        help='Reverses test cases order.',
-    )
-    parser.add_argument(
-        '--use-existing-database',
-        action='store_true',
-        default=False,
-        help="Don't create a test database. USE AT YOUR OWN RISK!",
-    )
-    parser.add_argument(
-        '-k', '--keepdb',
-        action='store_const',
-        const=True,
-        help="Preserves the test DB between runs.",
-    )
-    parser.add_argument(
-        '-S', '--simple',
-        action='store_true',
-        default=False,
-        help="Use simple test runner that supports Django's"
-        " testing client only (no web browser automation)"
-    )
+    kwargs['dest'] = 'value'
+    parser = CommandParser(add_help=False, allow_abbrev=False)
+    parser.add_argument(*args, **kwargs)
+    try:
+        options, _ = parser.parse_known_args(argv[2:])
```

Comment:
This is hard to read. (e.g. What is the purpose of the slicing? Ignore the remaining arguments?)